Add a debugging example using Golang IDE

adibrastegarnia opened the pull request:
This PR adds an example of debugging of onos-config using Goland IDE to address issue #547

Andrea-Campanella commented on docs/debugging.md:
```diff
@@ -1,7 +1,16 @@
+- In this example, we assume you already started an onit cluster. If you haven't done that yet, refer to [testing](testing.md) for more details.
+
+- First, you need to configure your [Goland](https://www.jetbrains.com/go/) IDE to connect to the remote Delve debugger which is running in the onos-config 
+pod. You can do that by going to *Run | Edit Configurations… | + | Go Remote* and configuring the host (in this example *localhost*) and port (in this example port **40000**) your remote debugger is listening on.
```
is this port fixed or does it change every time ? if so is there a way to get it ?

adibrastegarnia replied:
This is the port that is used in the debug image to start the dlv server and it is fixed.

adibrastegarnia commented:
@kuujo @Andrea-Campanella 

This PR is ready for review. If you are looking for a specific debugging feature that should be documented, please let me know. I tried to cover the most important ones.

Andrea-Campanella commented on docs/debugging.md:
```diff
@@ -1,239 +1,297 @@
+                 
+- As you can see above, after setting the breakpoints, you should be able to see some messages
+in the onos-config pod logs that confirm you set the breakpoints. When we add a simulator
+the Goland IDE points to the first breakpoint when is hit and you can go through the list 
```
I'd reformulate with `the Goland IDE hits the first breakpoint that was set. At this point you can go....`

Andrea-Campanella commented on docs/debugging.md:
```diff
@@ -1,239 +1,297 @@
+the Goland IDE points to the first breakpoint when is hit and you can go through the list 
+of variables and goroutines using the debugger dashboard. 
+
+- Suppose we want to have tracepoints instead breakpoints. That means they don't 
```
missing `of` before breakpoints

Andrea-Campanella replied:
substitute `That means they` with `tracepoints don't cause.....`

adibrastegarnia commented:
> Awesome work! Minor changes requested.

Andrea, Thanks for catching the typos and mistakes. I think I addressed your comment. Please take a look and let me know if more changes needed.

adibrastegarnia commented:
@kuujo 

Whenever you get a chance take a look and then we merge.
